src/content/docs/en/guides/framework-components.mdx

```diff
@@ -121,13 +121,16 @@ import Counter from '../components/Counter.svelte';
 </div>
 ```
 
-:::caution[Passing functions as props]
-You can pass a function as a prop to a framework component, but it only works during server rendering. If you try to use the function in a hydrated component (for example, as an event handler), an error will occur.
 
-This is because functions can't be _serialized_ (transferred from the server to the client) by Astro.
-:::
 
 
+Props that are passed to interactive framework components [using a `client:*` directive](/en/reference/directives-reference/#client-directives) must be [serialized](https://developer.mozilla.org/en-US/docs/Glossary/Serialization): translated into a format suitable for transfer over a network, or storage. However, for performance and/or security reasons, Astro does not serialize every type of data structure. Therefore, there are some limitations on what can be passed as props to hydrated components.
+
+The following prop types are supported: 
+`plain object`, `number`, `string`, `Array`, `Map`, `Set`, `RegExp`, `Date`, `BigInt`, `URL`, `Uint8Array`, `Uint16Array`, `Uint32Array`, and `Infinity`
```

Comment:
```suggestion
plain object, `number`, `string`, `Array`, `Map`, `Set`, `RegExp`, `Date`, `BigInt`, `URL`, `Uint8Array`, `Uint16Array`, `Uint32Array`, and `Infinity`
```
The others are actual types, but "plain object" isn't. It could be "plain `object`" but I think it's better without it.